Cabal/doc/nix-local-build.rst

```diff
@@ -478,6 +478,36 @@ and caches if the ``--save-config`` option is given, in which case it only remov
 the build artefacts (``.hi``, ``.o`` along with any other temporary files generated
 by the compiler, along with the build output).
 
+cabal new-sdist
+---------------
+
+``cabal new-sdist [FLAGS] [TARGETS]`` takes the crucial files needed to build ``TARGETS``
+and puts them into an archive format ready for upload to Hackage. These archives are stable
+and two archives of the same format built from the same source will hash to the same value.
```

Comment:
For `autogen-modules` we can't guarantee this 100% since we have no control over non-determinism in preprocessors, especially custom ones. For example, a preprocessor can add a comment with the date when the file was generated, or have output that depends on system locale settings. I guess that we can settle on saying that the behaviour is undefined when a preprocessor is non-deterministic in a way that can be observed from compiled code.

Reply:
`autogen-modules` *shouldn't* (and it's a bug if they are, that  probably needs a test to be sure) be included for this very reason.

Reply:
You're correct! Sorry, it's getting a bit late here.